habitat-lab/habitat/tasks/nav/nav.py

```diff
@@ -1040,6 +1040,24 @@ def update_metric(
         self._previous_distance = distance_to_target
 
 
+class NavigationMovementAgentAction(SimulatorTaskAction):
+    def __init__(self, *args, config, sim, **kwargs):
+        super().__init__(*args, config=config, sim=sim, **kwargs)
+        self._sim = sim
+        self._tilt_angle = config.tilt_angle
+
+    def _move_camera_vertical(self, amount: float):
+        assert (
+            len(self._sim.agents) == 1  # type: ignore
+        ), "For navigation tasks, there can be only one agent in the scene"
+        sensor_names = list(self._sim.agents[0]._sensors.keys())  # type: ignore
```

Comment:
Why assign a temporary list here? Just iterate through the keys / dict directly?

Reply:
I am changing the dictionary in the for loop. You have to make a copy of the keys to be able to iterate.